regolith/tools.py

```diff
@@ -1287,4 +1289,31 @@ def get_id_from_name(coll, name):
                              case_sensitive=False)
     if person:
         return person["_id"]
-    else: return None
\ No newline at end of file
+    else: return None
+
+def is_fully_appointed(appts, begin_date, end_date):
```

Comment:
do we want to pass a person rather than a list of appointments?  Grammatically, ``is_fully_appointed(aejaz)`` seems to make more sense to me.

then logically it should return True or False.

I could imagine it taking dates as optional args.  default values for these could be the begin_date of the student (not the appointment) and end_date if it exists with now as default?  We may need a ``supported: True`` flag in the ``employment`` item to get meaningful begin and end dates for the student.   Maybe too brittle, I am not sure.  Easier would be to take as default begin and end dates as the earliest and the latest date in the list of appointments.

Reply:
I like that your function identifies when the the person is over/under appointed.  Can we simply print this to stdOut as a warning rather than returning it as a value?

Reply:
I have changed it to take in a `person` and added default behaviour for the dates (earliest and latest appointment date)
As for the second thing - I think that is more elegant, but the problem is that it then prints a min every time the function is called in the helper file, which is a problem because then we have to move the `print(appointment)` into the loop, and the stdOut will display
`min 1.0 at 2012-12-32`
`aejaz f21`
i.e. the appointment below the minimum and not above, which is strange to read... and maybe some other problems I can't anticipate. do you have any suggestions for how to handle this?

Reply:
not specifically but isn't that the behavior that I want?

From my perspective, it is helpful to know which appointments need to be adjusted because of an error in loading or grant etc. so if I get the person_id and the appointment_id of the bad appointment I can go and find it.    I guess if there is a gap in appointments, knowing the appointment_id of the preceding and following appointments would help me zoom into it.

Reply:
``min 1.0 at 2012-12-31`` is not a very helpful error message either tbn....

Reply:
I see what you mean: I returned a list earlier so that my code would display something like
`person: MC Escher       appointment: f20`
`min 0.4 at 2020-09-01`
`person: Kurt Godel       appointment: ss20`
`max 1.3 at 2020-09-01`
And changing the function to return True/False and print the message will display
`min 0.4 at 2020-09-01`
`person: MC Escher       appointment: f20`
`max 1.3 at 2020-09-01`
`person: Kurt Godel       appointment: ss20`
So the order of the display changes since the function is called before the appointment is printed.
I will make the error message more verbose so hopefully that's more helpful!

Reply:
done! I have manipulated it to print both on the same line